### ISSUE TITLE
[Backend] Endpoint to retrieve institution Yellow Ribbon data

### ISSUE THREAD

**Issue by kelsonic** [department-of-veterans-affairs/va.gov-team#5567]
## User Story
As a Veteran, I want specific information about Yellow Ribbon programs so that I can make informed decisions about institutions I apply to. 

## Description

The [current Yellow Ribbon MVP spec](https://vsateams.invisionapp.com/share/X5VOMBG2YP4#/screens/402038362) requires 4 key-value pairs that are not currently found within [the GIDS endpoint](https://dev.va.gov/gids/v0/institutions). Those key-value pairs are `degree_level`, `division_professional_school`, `number_of_students`, and `contribution_amount`. The reason for this is that the endpoint is surfacing `Institution` models, not necessarily `YellowRibbonProgram` models.

As a result, and after discussion with the BAH team, we've decided to create a new endpoint (`https://va.gov/gids/v0/yellow-ribbon-programs`). The response from this endpoint **must include** an array of objects that have _at least_ the following key-value pairs:

```
 availableOnline
 city
 country
 id
 name
 onlineOnly
 state
 degreeLevel
 divisionProfessionalSchool
 numberOfStudents
 contributionAmount
```

Later, we may also implement pagination, `sortBy` + `sortDirection` optional query params, and key-level filtering (e.g. `&degreeLevel=undergraduate` query param).

## Acceptance Criteria
- [ ] Create `GET https://va.gov/gids/v0/yellow-ribbon-programs` that returns a 200 response in the shape we expect (see above for key-value pairs).
- [ ] Add error handling and alternate response codes.
- [ ] Add pagination.
- [ ] Add `sortBy` and `sortDirection` optional query params.
- [ ] Add the above keys as query params for filtering.
- [ ] Write unit tests for all code pertaining to the PR.

**Comment by MickinSahni** [comment]
Thanks, Kelson. I'm moving this to Backlog until you and Nick have a chance to break this down. One thing to consider has you break this down -- does this warrant conversion into an Epic? It might make sense for an 8-pointer so we're not carrying this ticket thru a bunch of sprints. @kelsonic @ncksllvn

**Comment by ncksllvn** [comment]
Recommendation for how to break this down -

1. Create GIDS endpoint to expose Yellow Ribbon data 
2. Add support for pagination to GIDS endpoint (blocked on 1)
3. Connect Vets-API to GIDS endpoint for Yellow Ribbon data (blocked on 2)
4. Add support for sorting to GIDS endpoint (blocked on 1)
5. Add support for filters to GIDS endpoint (blocked on 1)
6. [Front-end] Implement new Yellow Ribbon API route into front-end React app (blocked on 3)

On 4, 5 - We should confirm that these are hard requirements for MVP before starting, along with associated specs - what fields we need to support sorting and filtering.

**Comment by kelsonic** [comment]
This ticket can now be closed.

@ncksllvn Thanks for breaking those down! I created the following tickets that match with your list above:
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5597
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5598
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5599
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5602
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5604
1. https://github.com/department-of-veterans-affairs/va.gov-team/issues/5608

**Comment by kelsonic** [comment]
@ncksllvn Number 5 may actually be necessary for MVP. I think @andaleliz said that 4 is not needed for the MVP. Is 2 actually necessary for the MVP technically?

**Comment by ncksllvn** [comment]
You rock @kelsonic!

**Comment by andaleliz** [comment]
@kelsonic I don't know how to answer your question about #2.  I mean, technically, if we go by the AC written for the MVP, I guess not, it doesn't specifically call out pagination.  But, from a UX perspective, I think pagination is better than lazy load for this use case (searching vs casual browse). And I wouldn't want to just straight up return the whole search result in one go as I imagine that'd be slow and terrible, and a lot of data the user isn't actually interested in if they just want a few schools.

And yeah, if we have to pick sorting OR filtering, I'd be in favor of a more robust filter set over sorting options.
